File: core/src/com/gmaniliapp/flappyhermes/sprite/Sandal.java
package com.gmaniliapp.flappyhermes.sprite;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Sound;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector3;
import com.badlogic.gdx.utils.Array;

public class Sandal {

    private static final int GRAVITY = -15;
    private static final int SPEED = 100;

    private Vector3 position;
    private Vector3 velocity;

    private Array<Texture> frames;

    private int frameCount;
    private int currentFrameIndex;

    private float maxFrameTime;
    private float currentFrameTime;

    private Rectangle bounds;

    private Sound wing;

    public Sandal(int x, int y) {
        position = new Vector3(x, y, 0);

        velocity = new Vector3(0, 0, 0);

        frames = new Array<>();
        frames.add(new Texture("sandal_downflap.png"));
        frames.add(new Texture("sandal_midflap.png"));
        frames.add(new Texture("sandal_upflap.png"));

        frameCount = frames.size;
        currentFrameIndex = 0;
        maxFrameTime = 0.5f / frameCount;

        bounds = new Rectangle(x, y, getTexture().getWidth(), getTexture().getHeight());

        wing = Gdx.audio.newSound(Gdx.files.internal("wing.ogg"));
    }

    public void update(float deltaTime) {
        currentFrameTime += deltaTime;
        if (currentFrameTime > maxFrameTime) {
            currentFrameIndex++;
            currentFrameTime = 0;
        }
        if (currentFrameIndex >= frameCount) {
            currentFrameIndex = 0;
        }

        if (position.y > 0) {
            velocity.add(0, GRAVITY, 0);
        }

        velocity.scl(deltaTime);

        position.add(SPEED * deltaTime, velocity.y, 0);

        if (position.y < 0) {
            position.y = 0;
        }

        velocity.scl(1 / deltaTime);

        bounds.setPosition(position.x, position.y);
    }

    public void jump() {
        velocity.y = 300;
        wing.play();
    }

    public void dispose() {
        for (int i = 0; i < frames.size; i++) {
            frames.get(i).dispose();
        }
        wing.dispose();
    }

    public Vector3 getPosition() {
        return position;
    }

    public Texture getTexture() {
        return frames.get(currentFrameIndex);
    }

    public Rectangle getBounds() {
        return bounds;
    }
}

File: core/src/com/gmaniliapp/flappyhermes/sprite/Sandal.kt
package com.gmaniliapp.flappyhermes.sprite

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.audio.Sound
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.math.Vector3
import com.badlogic.gdx.utils.Array

class Sandal(x: Int, y: Int) {

    private val velocity: Vector3 = Vector3(0f, 0f, 0f)
    private val frames: Array<Texture> = Array()
    private val frameCount: Int
    private var currentFrameIndex: Int
    private val maxFrameTime: Float
    private var currentFrameTime = 0f
    private val wing: Sound

    val position: Vector3 = Vector3(x.toFloat(), y.toFloat(), 0f)

    val bounds: Rectangle

    fun update(deltaTime: Float) {
        currentFrameTime += deltaTime
        if (currentFrameTime > maxFrameTime) {
            currentFrameIndex++
            currentFrameTime = 0f
        }
        if (currentFrameIndex >= frameCount) {
            currentFrameIndex = 0
        }
        if (position.y > 0) {
            velocity.add(0f, GRAVITY.toFloat(), 0f)
        }
        velocity.scl(deltaTime)
        position.add(SPEED * deltaTime, velocity.y, 0f)
        if (position.y < 0) {
            position.y = 0f
        }
        velocity.scl(1 / deltaTime)
        bounds.setPosition(position.x, position.y)
    }

    fun jump() {
        velocity.y = 300f
        wing.play()
    }

    fun dispose() {
        for (i in 0 until frames.size) {
            frames[i].dispose()
        }
        wing.dispose()
    }

    val texture: Texture
        get() = frames[currentFrameIndex]

    companion object {
        private const val GRAVITY = -15
        private const val SPEED = 100
    }

    init {
        frames.add(Texture("sandal_downflap.png"))
        frames.add(Texture("sandal_midflap.png"))
        frames.add(Texture("sandal_upflap.png"))
        frameCount = frames.size
        currentFrameIndex = 0
        maxFrameTime = 0.5f / frameCount
        bounds = Rectangle(x.toFloat(), y.toFloat(), texture.width.toFloat(), texture.height.toFloat())
        wing = Gdx.audio.newSound(Gdx.files.internal("wing.ogg"))
    }
}